File: leetcode/1. Two Sum/main.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {
	prevMap := make(map[int]int)

	for i, n := range nums {
		diff := target - n
		if _, containsKey := prevMap[diff]; containsKey {
			return []int{prevMap[diff], i}
		}
		prevMap[n] = i
	}
	fmt.Println(target)
	fmt.Println(nums)
	return []int{}
}

func main() {
	input := []int{2, 7, 11, 15}

	keyPair := twoSum(input, 9)

	fmt.Println(keyPair)
}

File: leetcode/1. Two Sum/main.py
#!/bin/env python

def main():
    print("Hello World!")

    print(twoSome([2,7,11,15], 9))

def twoSome(nums: list[int], target: int) -> list[int]:
    """
    Given an array of integers nums and an integer target, return indices of the two numbers such that they add up to target.
    You may assume that each input would have exactly one solution, and you may not use the same element twice.
    You can return the answer in any order.
    """
    for i in range(len(nums)):
        for j in range(i + 1, len(nums)):
            if nums[i] + nums[j] == target:
                return [i, j]
    return []

if __name__ == "__main__":
    main()